### PR TITLE
Add the geo_location field to assets

### PR DESCRIPTION
**Pull request by tommythorsen**
I have moved the related code out of files, into shared, so that it can
be used by both files and assets.

## Description
Please describe the change you have made.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change. 
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring. 
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by arnels — APPROVED** [review]

**Comment by tommythorsen** [comment]
On second thought, we may want to hold off on this until the new `geoLocation` field is in the service contract.

That will be done by this PR: https://github.com/cognitedata/service-contracts/pull/1746

**Review by tuanng-cognite — APPROVED** [review]

**Comment by arnels** [comment]
@cognitedata/starbase Your review is required.

File: cognite/client/data_classes/assets.py [hunk 1]
```diff
@@ -268,6 +275,10 @@ def parent_external_id(self):
     def labels(self):
         return AssetUpdate._LabelAssetUpdate(self, "labels")
 
+    @property
+    def geoLocation(self):
```
Comment (erlendvollset) [inline cognite/client/data_classes/assets.py:279]: we prefer snake_case over camelCase
```suggestion
    def geo_location(self):
```

**Review by erlendvollset — APPROVED** [review]
LGTM. Just need to address the name of the AssetUpdate property

**Pull request merged**